pages/_conferenceCode/contacts/index.vue

```diff
@@ -346,48 +360,120 @@ function initData (){
     },
     meetingsOptions         : [],
     organizationOptions     : [],
-    organizationTypesOptions: []
+    organizationTypesOptions: [],
+    searchTagInput          : '',
+    countryList             : []
   }
 }
 
-function isFiltersApplied (_this){
+// ===================
+// Mounted method
+// ===================
+async function mounted (){
+  const qs = this.$route.query
+
+  this.meetingsOptions = await getMajorMinorMeetings.apply(this)
+
+  this.countryList = this.$store.getters['countries/asSelectOptions']
+
+  sortCountryByAsc.apply(this)
+
+  this.organizationTypesOptions = this.$store.getters[
+    'organizations/asSelectOptions'
+  ]
+
+  await setQueryStringFilters.call(this, qs)
+}
+
+function isFiltersApplied (){
   if (
-    _this.filter.name ||
-    _this.filter.isBroadSearch ||
-    _this.filter.selectedMeetings.length ||
-    _this.filter.selectedOrganizationType ||
-    _this.filter.selectedOrganization.length ||
-    _this.filter.selectedCountry.length ||
-    _this.filter.selectedFlag.length ||
-    _this.filter.selectedAttendance.length ||
-    _this.filter.selectedScop
+    this.filter.name ||
+    this.filter.isBroadSearch ||
+    this.filter.selectedMeetings.length ||
+    this.filter.selectedOrganizationType ||
+    this.filter.selectedOrganization.length ||
+    this.filter.selectedCountry.length ||
+    this.filter.selectedFlag.length ||
+    this.filter.selectedAttendance.length ||
+    this.filter.selectedScop
   )
     return true
   else return false
 }
 
-async function mounted (){
-  this.meetingsOptions = await getMajorMinorMeetings(this)
+async function organizationSearchChangeEvent (query){
+  this.isLoadingOrganization = true
+  if (query)
+    this.organizationOptions = await this.$kronosApi.getOrganizations({
+      FreeText: query,
+      limit   : 25
+    })
 
-  this.organizationOptions = await this.$kronosApi.getOrganizations({
-    limit: 5000
-  })
+  this.isLoadingOrganization = false
+}
+
+async function setQueryStringFilters (qs){
+  if (!_.isEmpty(qs)){
+    if (qs.FreeText) this.searchTagInput = qs.FreeText
+    selectSelectedCountry.call(this, qs.Governments)
+    selectSelectedMeetings.call(this, qs.EventUIDs)
+    selectCountryScope.call(this, qs.CountryScope)
+    selectRegistrationStatus.call(this, qs.EventRegistrationStatus)
+    await selectOrganizations.call(this, qs.OrganizationUIDs)
+  }
+}
 
-  const country = await this.$kronosApi.getCountries()
+async function selectOrganizations (organizationUIDs){
+  if (organizationUIDs){
+    this.organizationOptions = await this.$kronosApi.getOrganizations({
+      OrganizationUIDs: asArray(organizationUIDs)
+    })
+    this.filter.selectedOrganization = asArray(organizationUIDs).map(o =>
+      this.organizationOptions.find(x => x.OrganizationUID === o)
+    )
+  }
+}
```

Comment:
I think we should change the behaviour of the function.  if I pass falsey / empty array the selection should be cleared. this will also remove the  necessity of having a separate clear function. 

we should apply this behaviour to all select/set field function.  

selectedOrganization is multiple so should be plural: `selectedOrganizations`

the way you use map & find is confusing.  better to use find & includes. *find all orgs their Id is included in the uids list*

```
 organizationOptions.find(o => uids.includes(o.OrganizationUID)
```


```suggestion
async function selectOrganizations (organizationUIDs){
  if (organizationUIDs && organizationUIDs.length) {
    this.organizationOptions = await this.$kronosApi.getOrganizations({
      OrganizationUIDs: asArray(organizationUIDs)
    })
    let uids =  asArray(organizationUIDs)
    this.filter.selectedOrganizations= this.organizationOptions.find(o => uids.includes(o.OrganizationUID))
    )
  }
  else {
    this.organizationOptions = []
    this.filter.selectedOrganizations = []
  }
}
```